src/uart.rs

```diff
@@ -0,0 +1,375 @@
+    /// uart.set_mode(DataBits::Eight, Parity::None, StopBits::One)   
+    /// .unwrap_or_else(|e| panic!("Error setting UART mode: {e:?}"));
+    /// ```
+    #[cfg(riot_module_periph_uart_modecfg)]
+    pub fn set_mode(
+        &mut self,
+        data_bits: DataBits,
+        parity: Parity,
+        stop_bits: StopBits,
+    ) -> Result<(), UartDeviceError> {
+        unsafe {
+            match UartDeviceError::from_c(uart_mode(
+                self.dev,
+                data_bits.to_c(),
+                parity.to_c(),
+                stop_bits.to_c(),
+            )) {
+                UartDeviceError::Success => Ok(()),
+                status => Err(status),
+            }
+        }
+    }
+
+    /// Transmits the given data via the `UART`-device
+    ///
+    /// # Examples
+    /// ```
+    /// use riot_wrappers::uart::UartDevice;
+    /// let mut uart = UartDevice::new_without_rx(0, 115200)
+    ///    .unwrap_or_else(|e| panic!("Error initializing UART: {e:?}"));
+    /// uart.write(b"Hello from UART\n");
+    /// ```
+    pub fn write(&mut self, data: &[u8]) {
+        unsafe {
+            uart_write(self.dev, data.as_ptr(), data.len() as size_t);
+        }
+    }
+
+    /// Turns on the power from the `UART-Device`
+    pub fn power_on(&mut self) {
+        unsafe { uart_poweron(self.dev) };
+    }
+
+    /// Turns off the power from the `UART-Device`
+    pub fn power_off(&mut self) {
+        unsafe { uart_poweroff(self.dev) };
+    }
+
+    /// This function normally does not need to be called. But in some case, the pins on the `UART`
+    /// might be shared with some other functionality (like `GPIO`). In this case, it is necessary
+    /// to give the user the possibility to init the pins again.
+    #[cfg(riot_module_periph_uart_reconfigure)]
+    pub unsafe fn init_pins(&mut self) {
+        uart_init_pins(self.dev);
+    }
+
+    /// Change the pins back to plain GPIO functionality
+    #[cfg(riot_module_periph_uart_reconfigure)]
+    pub unsafe fn deinit_pins(&mut self) {
+        uart_deinit_pins(self.dev);
+    }
+
+    /// Get the RX pin
+    #[cfg(riot_module_periph_uart_reconfigure)]
+    pub fn get_pin_rx(&mut self) -> Option<crate::gpio::GPIO> {
+        crate::gpio::GPIO::from_c(unsafe { uart_pin_rx(self.dev) })
+    }
+
+    /// Get the TX pin
+    #[cfg(riot_module_periph_uart_reconfigure)]
+    pub fn get_pin_tx(&mut self) -> Option<crate::gpio::GPIO> {
+        crate::gpio::GPIO::from_c(unsafe { uart_pin_tx(self.dev) })
+    }
+
+    /// Configure the function that will be called when a start condition is detected
+    /// This will not enable / disable the generation of the RX start interrupt
+    /// # Arguments
+    /// * `user_fxopt` - The user defined callback function that gets called when a start condition is detected
+    #[cfg(riot_module_periph_uart_rxstart_irq)]
+    pub fn rxstart_irq_configure<F>(&mut self, user_fxopt: &'a mut F)
```

Comment:
Why the 'a? It's probably fine to just have this for 'static functions, but then the function needs to be moved in here, and not taken briefly by exclusive reference.

If you have no concrete use case for this in your application, it may make sense to remove this function and add it later on demand, when we have a good understanding of what typical requirements are.

Reply:
I have to admit, I did not have a use case for this function(s), so I removed them completely.